addons/openebs/2.6.0/install.sh

```diff
@@ -253,13 +259,247 @@ function openebs_upgrade() {
 
     local runningVer
     runningVer=$(kubectl -n $OPENEBS_NAMESPACE get deploy openebs-provisioner -o jsonpath='{.metadata.labels.openebs\.io/version}')
+    export PREVIOUS_OPENEBS_VERSION=$runningVer
 
     local semVerRunningList=( ${runningVer//./ } )
-    local semVerInstallList=( ${OPENEBS_VERSION//./ } )
 
-    if [[ ${semVerRunningList[0]} -lt 2 ]]; then
-        bail "Upgrades from 1.x versions of openebs are not yet supported."
+    if [[ ${semVerRunningList[0]} -eq 2 ]] && [[ ${semVerRunningList[1]} -eq 6 ]]; then
+      # this is already up to date
+      return 0
     fi
 
+    if [[ ${semVerRunningList[0]} -lt 2 ]] && [[ ${semVerRunningList[1]} -lt 12 ]]; then
+        bail "Upgrades from pre-1.12 versions of openebs are not yet supported."
+    fi
+
+    openebs_spc_to_cspc
+}
+
+function openebs_cspc_upgrade() {
+    # start upgrade process - see https://github.com/openebs/upgrade/blob/v2.6.0/docs/upgrade.md
+    kubectl delete csidriver cstor.csi.openebs.io || true
+
+    openebs_upgrade_pools
     return 0
 }
+
+function openebs_cleanup_kubesystem() {
+  # cleanup old kube-system statefulsets
+  # https://github.com/openebs/upgrade/blob/v2.6.0/docs/upgrade.md#prerequisites-1
+  kubectl -n kube-system delete sts openebs-cstor-csi-controller || true
+  kubectl -n kube-system delete ds openebs-cstor-csi-node || true
+  kubectl -n kube-system delete sa openebs-cstor-csi-controller-sa,openebs-cstor-csi-node-sa || true
+}
+
+function openebs_spc_to_cspc() {
+    # upgrade job from https://github.com/openebs/upgrade/blob/v2.6.0/examples/migrate/spc-migration.yaml
+
+    local pools
+    pools=$(kubectl get spc --template '{{range .items}}{{.metadata.name}}{{"\n"}}{{end}}')
+
+    for pool in $pools; do
+        local spcCurrentVer=$(kubectl get spc $pool -o jsonpath='{.versionDetails.status.current}')
+        logSubstep "Upgrading $pool from SPC to CSPC"
+        local out_file=/tmp/openebs-spc-cspc-$pool.yaml
```

Comment:
Should we put this in the kurl install directory (three affected files here)? No strong opinion.

Reply:
NM, I just realized these files are tmp not var.